contribute/README.md

```diff
@@ -10,9 +10,13 @@ Thanks for enhancing OpenFisca anyway!
 
 ## Why contribute to OpenFisca?
 
-OpenFisca is a project being developed under the GPLv3 license or later.
+OpenFisca is a project being developed under the AGPL-3.0 license.
 The source code is freely available and modifiable.
 
+We ask for improvements and evolution of OpenFisca to be shared back, for it allows the community to grow through improvements to the _core engine_ and updates to the _modelled tax-benefit systems_ (country packages, extensions, reforms, etc.).
+
+Although, as long as the software you build with OpenFisca can't be considered an improvement or an evolution of OpenFisca, we won't consider it a form of derivative work. If the latter holds true, we won't ever ask you to share your software's source code, whether you're using the Python API, the Web API or both.
```

Comment:
Continuing a previous [discussion](https://github.com/openfisca/openfisca-doc/pull/117#discussion_r176851980) that has been marked as outdated but still sounds relevant to me:

Let's not confuse legally-binding stuff, intentions and actual usage.

Under a copyleft license, reusers are **legally bound** to redistribute their work.

Now, if you don't do it, no lightning is going to strike you. What happens is that if your **intention** is to free-ride and it does negatively impact the ecosystem, then the contributors have leverage to make you open your code. If you do abide by the license just for legalities because you've developed a service and slightly tweaked the code for it but it has no added value for the ecosystem, no one is going to care ([I know](https://github.com/MattiSG/dokuwiki-template-vector)).

If you have an unfinished reform that no one cares about, how would I even know you're doing that and you're not abiding by the license? You are doing something illegal by not sharing all your changes, but no one cares, or even knows.

However, we should **never** write a provision that says you're allowed not to open until it is “finished”. Otherwise, our leverage is screwed: when we realise you've written a 100x more efficient `net_to_brut` reform and we see it because other reusers have their service at risk due to unfair competition after they've invested 10 months in improving the legislation model, you can simply claim that “oh but it's experimental, it's not finished yet, I can't open it right now”.

To give another metaphor, the law doesn't allow you to cross a street if the light is red, even if its 2 in the morning and there is not a single car around. It is your responsibility to cross or not, and 99.9999% of the time no one will ever care —or even know.
But if you *do* kill the driver of a car that does an emergency brake to avoid you, you are responsible for your action. That's what laws (and licenses, for that matter) are for.

I'm sure you've all copied a sentence or even a paragraph from Wikipedia without citing the source and its license. Yet it's critical that Wikipedia doesn't allow you to do that under any circumstances, because it does not open loopholes for publishers.

Reply:
> Under a copyleft license, reusers are legally bound to redistribute their work.

Yes! Although, there are parts of the licence that have been subject of dispute and needed further clarification. It is the case with both OBbL and AGPL-3.0. The objective is to clarify the scope of the virality and add particular terms and conditions.

> Now, if you don't do it, no lightning is going to strike you. What happens is that if your intention is to free-ride and it does negatively impact the ecosystem, then the contributors have leverage to make you open your code. If you do abide by the license just for legalities because you've developed a service and slightly tweaked the code for it but it has no added value for the ecosystem, no one is going to care.

Sure, the objective here is to incentive people who wouldn't use the software because of the licence to say "Hey, if you're creating your simulator, you don't have to share your app's code." That is more or less clear with a Web API, but no so much with the Python API, as depending on the interpretation it could be considered a derivative work, and see virality of the licence applied to it (or not).

> If you have an unfinished reform that no one cares about, how would I even know you're doing that and you're not abiding by the license? You are doing something illegal by not sharing all your changes, but no one cares, or even knows.

I'm not sure, as licence applies to distribution of the software (like running it as a service). Reforms are a tricky question.

> However, we should never write a provision that says you're allowed not to open until it is “finished”. Otherwise, our leverage is screwed: when we realise you've written a 100x more efficient net_to_brut reform and we see it because other reusers have their service at risk due to unfair competition after they've invested 10 months in improving the legislation model, you can simply claim that “oh but it's experimental, it's not finished yet, I can't open it right now”.

👍